File: camerax_android/android/src/main/kotlin/dev/hebei/camerax_android/core/ExposureStateImpl.kt
package dev.hebei.camerax_android.core

import androidx.camera.core.ExposureState
import dev.hebei.camerax_android.CameraXImpl
import dev.hebei.camerax_android.common.IntRange
import dev.hebei.camerax_android.PigeonApiExposureStateApi

class ExposureStateImpl(impl: CameraXImpl) : PigeonApiExposureStateApi(impl) {
    override fun exposureCompensationIndex(pigeon_instance: ExposureState): Long {
        return pigeon_instance.exposureCompensationIndex.toLong()
    }

    override fun exposureCompensationRange(pigeon_instance: ExposureState): IntRange {
        val obj = pigeon_instance.exposureCompensationRange
        return IntRange(obj)
    }

    override fun exposureCompensationStep(pigeon_instance: ExposureState): Double {
        return pigeon_instance.exposureCompensationStep.toDouble()
    }

    override fun isExposureCompensationSupported(pigeon_instance: ExposureState): Boolean {
        return pigeon_instance.isExposureCompensationSupported
    }
}

File: camerax_ios/lib/src/core/exposure_state_impl.dart
// import 'package:camerax_ios/src/api.dart';
// import 'package:camerax_ios/src/common.dart';
// import 'package:camerax_platform_interface/camerax_platform_interface.dart';

// final class ExposureStateImpl implements ExposureState {
//   final ExposureStateProxyApi api;

//   ExposureStateImpl.internal(this.api);

//   @override
//   int get exposureCompensationIndex => api.exposureCompensationIndex;
//   @override
//   Range<int> get exposureCompensationRange =>
//       api.exposureCompensationRange.impl;
//   @override
//   double get exposureCompensationStep => api.exposureCompensationStep;
//   @override
//   bool get isExposureCompensationSupported =>
//       api.isExposureCompensationSupported;
// }

// extension ExposureStateProxyApiX on ExposureStateProxyApi {
//   ExposureState get impl => ExposureStateImpl.internal(this);
// }
